infra/security.tf

```diff
@@ -52,7 +52,7 @@ module "iam_assumable_role_with_oidc" {
 
   number_of_role_policy_arns = 1
 
-  role_policy_arns = [module.iam_policy.arn]
+  role_policy_arns = [module.iam_policy.arn, "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"]
```

Comment:
remove this additional policy that you added for now, we will see later which permissions are required for deployment by GitHub.

Also remove this "depends_on = [module.iam_policy]" as this module already has implicit dependency on this line.